File: copilot-demo-labs/use-cases/03-code-explanation-docs/src/LegacyParser.py

def parse_config(lines):
    data = {}
    section = None
    for raw in lines:
        line = raw.strip()
        if not line or line.startswith('#'):
            continue
        if line.startswith('[') and line.endswith(']'):
            section_name = line[1:-1]
            section = {}
            data[section_name] = section
            continue
        idx = line.find('=')
        if idx > 0:
            key = line[:idx].strip()
            value = line[idx+1:].strip()
            if section is None:
                data[key] = value
            else:
                section[key] = value
    return data

def run_lab_variant(payload: dict) -> dict:
    # Expects payload['lines'] to be a list of strings
    if not payload or 'lines' not in payload or not isinstance(payload['lines'], list):
        raise ValueError("Payload must have a 'lines' list")
    return parse_config(payload['lines'])

File: copilot-demo-labs/use-cases/03-code-explanation-docs/src/LegacyParser.cs
namespace Demo;

public static class LegacyParser
{
    public static Dictionary<string, object> ParseConfig(IEnumerable<string> lines)
    {
        var data = new Dictionary<string, object>();
        Dictionary<string, string>? section = null;

        foreach (var raw in lines)
        {
            var line = raw.Trim();
            if (string.IsNullOrWhiteSpace(line) || line.StartsWith("#"))
            {
                continue;
            }

            if (line.StartsWith("[") && line.EndsWith("]"))
            {
                var sectionName = line[1..^1];
                section = new Dictionary<string, string>();
                data[sectionName] = section;
                continue;
            }

            var idx = line.IndexOf('=');
            if (idx > 0)
            {
                var key = line[..idx].Trim();
                var value = line[(idx + 1)..].Trim();
                if (section is null)
                {
                    data[key] = value;
                }
                else
                {
                    section[key] = value;
                }
            }
        }

        return data;
    }
}
